doc/charm++/manual.rst

```diff
@@ -8280,15 +8259,31 @@ timestep. To save resources, the PUP routine can well omit the 8
 intermediate matrices and checkpoint the matrix with the final results
 of each timestep.
 
-Group, nodegroup (Section :numref:`sec:group`) and singleton chare
+Group, nodegroup (Section :numref:`sec:group`), and singleton chare
 objects are normally not meant to be migrated. In order to checkpoint
 them, however, the user has to write PUP routines for the groups and
 chare and declare them as ``[migratable]`` in the .ci file. Some
 programs use *mainchares* to hold key control data like global object
-counts, and thus mainchares need to be checkpointed too. To do this, the
-programmer should write a PUP routine for the mainchare and declare them
-as ``[migratable]`` in the .ci file, just as in the case of Group and
-NodeGroup.
+counts, and thus mainchares need to be checkpointed too. To do this,
+the programmer should write a PUP routine for the mainchares and
+declare them as ``[migratable]`` in the .ci file, just as in the case
+of group and nodegroup.
+
+Variables marked as readonly are automatically checkpointed and
+restored by the runtime system, so user PUP routines do not need to
+explicitly handle them.
+
+Checkpointing variables of type ``CProxy_*`` is completely valid.
+After the restart callback is invoked, the original chare structures
+will have been reconstructed and all proxy variables will be valid
+references to the restarted versions of whatever they originally
+referred to. The only caveat to this is when the application is
+restarted on a different number of processors than it was checkpointed
+on, in which case non-location invariant chares have special behavior:
+singleton chares are not created or restored at all and
```

Comment:
question: is `mainchare` a `singleton` in this context?

Reply:
No, it's treated distinctly, I'll add a note to that effect.